src/IO/H5/VolumeData.cpp

```diff
@@ -159,6 +165,335 @@ void append_element_extents_and_connectivity(
+  sorted_element_indices.reserve(number_of_blocks);
+
+  // Properly sizes subvectors of sorted_element_indices
+  for (size_t i = 0; i < number_of_blocks; ++i) {
+    std::vector<size_t> sizing_vector;
+    auto number_of_elements_in_block =
+        static_cast<size_t>(std::count(block_number_for_each_element.begin(),
+                                       block_number_for_each_element.end(), i));
+    sizing_vector.reserve(number_of_elements_in_block);
+    sorted_element_indices.push_back(sizing_vector);
+  }
+
+  // Organizing grid_names by block
+  for (size_t i = 0; i < block_number_for_each_element.size(); ++i) {
+    sorted_element_indices[block_number_for_each_element[i]].push_back(i);
+  }
+
+  return std::make_tuple(number_of_blocks,
+                         std::move(block_number_for_each_element),
+                         std::move(sorted_element_indices));
+}
+
+// Sort input by block
+template <typename T>
+std::vector<std::vector<T>> sort_by_block(
+    const std::vector<std::vector<size_t>>& sorted_element_indices,
+    const std::vector<T>& property_to_sort) {
+  std::vector<std::vector<T>> sorted_property;
+  sorted_property.reserve(sorted_element_indices.size());
+
+  // Properly sizes subvectors
+  for (const auto& sorted_block_index : sorted_element_indices) {
+    std::vector<T> sizing_vector;
+    sizing_vector.reserve(sorted_block_index.size());
+    for (const auto& sorted_element_index : sorted_block_index) {
+      sizing_vector.push_back(property_to_sort[sorted_element_index]);
+    }
+    sorted_property.push_back(std::move(sizing_vector));
+  }
+
+  return sorted_property;
+}
+
+// Returns a std::tuple of the form
+// [expected_connectivity_length, expected_number_of_grid_points, h_ref_array],
+// where each of the quantities in the tuple is computed for each block
+// individually. expected_connectivity_length is the expected length of the
+// connectivity for the given block. expected_number_of_grid_points is the
+// number of grid points that are expected to be within the block. h_ref_array
+// is an array of the h-refinement in the x, y, and z directions. This function
+// computes properties at the block level, as our algorithm for constructing the
+// new connectivity works within a block, making it convenient to sort these
+// properties early.
+template <size_t SpatialDim>
+std::tuple<size_t, size_t, std::array<int, SpatialDim>>
+compute_block_level_properties(
+    const std::vector<std::string>& block_grid_names,
+    const std::vector<std::vector<size_t>>& block_extents) {
+  size_t expected_connectivity_length = 0;
+  size_t expected_number_of_grid_points = 0;
+
+  for (const auto& extents : block_extents) {
+    size_t element_grid_points = 1;
+    size_t element_connectivity_length = 1;
+    for (size_t j = 0; j < SpatialDim; j++) {
+      element_grid_points *= extents[j];
+      element_connectivity_length *= extents[j] - 1;
+    }
+    // Connectivity that already exists
```

Comment:
I found this and the next comment to not give me a good idea of the next line. How does the fact that a connectivity exists (and how do I know it exists?) relate to `+= element_connectivity_length * 8`? Maybe first answer here on GitHub and then I can make a concrete suggestion for the code?

It seems like `// Used for reserving the length of block_logical_coords` could be on line 254 to describe the variable, though.

Reply:
I took a closer look at this code and I realize that I should actually fix a couple of things.

What is happening in this chunk of the function is that we are computing by hand "how much" connectivity should exist without element-to-element connections. Natively we only have connectivity for cells. Each cell should contribute `2^SpatialDim` values to the connectivity since that is how many grid points are necessary to define a cell in a particular spatial dimension (i.e. in 3D, 8 grid points are connected as a hexahedron so this adds 8 to the length of the connectivity list). We are thus guaranteed to know that this connectivity must exist, otherwise, we have nothing to visualize. We then later compute the additional connections necessary and put this all together to get the `expected_connectivity_length`.

As for the things I intend to change:
1) `element_connectivity_length` --> `number_of_cells_in_element` since this is more descriptive
2) `expected_connectivity_length += element_connectivity_length * 8;` --> `expected_connectivity_length += element_connectivity_length * pow(2, SpatialDim);` since this should technically change depending on `SpatialDim` (this never errored before since `expected_connectivity_length` only reserves a vector length)

If there's anything else to change let me know!

Reply:
That sounds perfect! Thank you for the clarifications!